### PR TITLE
Fix alignof in ioctl.c to not use UB

### PR DESCRIPTION
**Pull request by tlively**
Defining types inside offsetof expressions is explicitly made UB in C2x and is
now rejected by clang. Use _Alignof instead since it is available in the version
of C we use to compile our libc.

### CODE REVIEW

**Comment by tlively** [comment]
Current dependencies on/for this PR:
* main
  * **PR #18510** <a href="https://app.graphite.dev/github/pr/emscripten-core/emscripten/18510" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/emscripten-core/emscripten/18510?utm_source=stack-comment).

File: system/lib/libc/musl/src/misc/ioctl.c [hunk 1]
```diff
@@ -9,7 +9,12 @@
 #include <endian.h>
 #include "syscall.h"
 
-#define alignof(t) offsetof(struct { char c; t x; }, x)
+#ifdef __EMSCRIPTEN__
+// The upstream version below is UB in C2x and rejected by clang.
+#define alignof(t) _Alignof(t)
+#else
+#define alignof(t) offsetof(struct { char c; t x; }, x) */
```
Comment (kripken) [inline system/lib/libc/musl/src/misc/ioctl.c:16]:
```suggestion
#define alignof(t) offsetof(struct { char c; t x; }, x)
```

**Review by kripken — APPROVED** [review]

**Review by sbc100 — APPROVED** [review]

**Comment by MaskRay** [comment]
Informed musl maintainer.

**Comment by tlively** [comment]
Thanks!

**Pull request merged**